### PR TITLE
Supporting sample sizes for multiple independently targeted contests

### PR DESCRIPTION
**Pull request by MorganLove**
**Description**

- Updated the types to reflect api changes
- Updated the review and launch stage to have multiple sample size selection boxes for the targeted contests
- Updated launch process to use the format expected by the api changes

**Testing**

- None yet, waiting for review

**Progress**

- It looks like it works, need to double check with @jonahkagan that it all makes sense.

### CODE REVIEW

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx [hunk 1]
```diff
@@ -9,28 +9,38 @@ import { ISidebarMenuItem } from '../../../Atoms/Sidebar'
...
   style: 'percent',
 })
 
-interface IStringSampleSize {
-  size: string
-  type: string | null
-  prob: number | null
+interface IStringSampleSizes {
+  [key: string]: {
+    size: string
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:36]: why use strings here? seems like numbers would be easier, since that's the API uses

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx [hunk 2]
```diff
@@ -126,6 +153,15 @@ const Review: React.FC<IProps> = ({ prevStage, locked, refresh }: IProps) => {
       j.ballotManifest.processing.status === FileProcessingStatus.PROCESSED
   ).length
 
+  const initialValues = targetedContests.length
+    ? targetedContests.map(c => ({
+        [c.id]: sampleSizeOptions[c.id] ? sampleSizeOptions[c.id][0].size : '',
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:158]: the backend guarantees that the keys of `sampleSizeOptions` will include all the targeted contest ids

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx [hunk 3]
```diff
@@ -134,141 +170,165 @@ const Review: React.FC<IProps> = ({ prevStage, locked, refresh }: IProps) => {
...
+          )
           setIsConfirmDialogOpen(true)
         }}
       >
         {({
           values,
           handleSubmit,
           setFieldValue,
-        }: FormikProps<{ sampleSize: string }>) => (
+        }: FormikProps<{ sampleSizes: { [key: string]: string }[] }>) => (
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:270]: why a list of objects? i think life might be simpler with a mapping from contest id to sample size, so `{ [contestId: string]: number }`

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx [hunk 4]
```diff
@@ -47,23 +57,31 @@ const Review: React.FC<IProps> = ({ prevStage, locked, refresh }: IProps) => {
...
   const history = useHistory()
   const [isConfirmDialogOpen, setIsConfirmDialogOpen] = useState(false)
 
   useEffect(() => {
     ;(async () => {
       try {
-        const { sampleSizes }: { sampleSizes: ISampleSizeOption[] } = await api(
+        const {
+          sampleSizes: options,
+        }: { sampleSizes: ISampleSizeOptions } = await api(
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:71]: what do you think about pulling this out into a hook and defining the type there? like the pattern we've been using for other API data sets

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx (hunk 1 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:36]: @jonahkagan Strings are what the form converts everything to, so it has a separate type for that so it's easier to track the conversion to strings going into the form and then remembering to convert back to numbers before sending it to the api.

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx (hunk 2 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:158]: @jonahkagan I might have been overcautious, but my concern here was that the call to the api to get the sample sizes will not have returned yet on the first run through this component. So `sampleSizeOptions` will only be the default `{}`. We're getting data from two asynchronous sources here, for the targeted contests and the sample sizes, and it needs both in order to create the `initialValues` value. Hence the double-layered ternary and fallback data.

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx (hunk 3 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:270]: @jonahkagan My concern was that we don't know the keys (the contest ids being the keys), and so having it in an array would make it more palatable for the form. But I ended up having to do weird stuff to get at it anyway, so I don't think it ended up simplifying it in the end. I'm going to try if it will work with just the object.

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx (hunk 4 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:71]: Probably a good idea. It would clean things up significantly and keep in pattern with everything else. Not sure why this one wasn't pulled out already, actually.

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx [hunk 5]
```diff
@@ -11,26 +11,25 @@ import useAuditSettings from '../../useAuditSettings'
 import useContests from '../../useContests'
 import { IContest, ISampleSizeOption } from '../../../../types'
 import useJurisdictions, { FileProcessingStatus } from '../../useJurisdictions'
-import { api, checkAndToast, testNumber } from '../../../utilities'
+import { testNumber } from '../../../utilities'
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:14]: Screen Shot 2020-07-20 at 1 07 23 PM

was playing around and triggered this error message (I think by typing a number and then deleting it). i think it's a pretty confusing message to a user

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx [hunk 6]
```diff
@@ -126,6 +93,22 @@ const Review: React.FC<IProps> = ({ prevStage, locked, refresh }: IProps) => {
...
 
+  const initialValues: IFormOptions = sampleSizeOptions
+    ? Object.keys(sampleSizeOptions).reduce(
+        (a, contestId) => ({
+          ...a,
+          [contestId]: sampleSizeOptions[contestId][0].size,
+        }),
+        {}
+      )
+    : {}
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:104]: what happens if two of the sample size options have the same size number? i think it would break the form.

this isn't a new issue - I think the same issue was there previous to this PR. makes me think that the data structure for sample sizes that the backend gives is not great. each sample size option should have some sort of key or id that you can use to uniquely identify it - the size number is not guaranteed to be uniquely identifying.

thoughts?

Comment (jonahkagan) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:104]: another potential advantage to that approach would be that you could add in an option with a `custom` key so that you wouldn't need a separate state value to hold the custom size

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx [hunk 7]
```diff
@@ -136,139 +119,166 @@ const Review: React.FC<IProps> = ({ prevStage, locked, refresh }: IProps) => {
...
+            Object.keys(sizes).reduce((a, contestId) => {
+              if (sizes[contestId] === 'custom') {
+                return {
+                  ...a,
+                  [contestId]: customSampleSizes[contestId],
+                }
+              }
+              return { ...a, [contestId]: sizes[contestId] }
+            }, {})
+          )
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:209]: i notice that you use this pattern a few times. I've seen it abstracted out into a helper function called `mapValues` in previous codebases I've worked on, since it comes up a lot. that would look like this:

```ts
const mapValues = (obj, fn) =>
  Object.keys(obj).reduce((a, key) => ({
    ...a,
    [key]: fn(obj[key], key)
  }), {})

// or this might be a faster implementation, not that it matters so much
const mapValues = (obj, fn) =>
  Object.fromEntries(Object.entries(obj).map((key, val) => fn(val, key)))
```

and then this bit could just be:

```ts
setSampleSizes(mapValues(sizes, (size, contestId) =>
  sizes[contestId] === 'custom' ? customSampleSizes[contestId] : sizes[contestId]
))
```

File: client/src/components/MultiJurisdictionAudit/Setup/Review/useSampleSizes.ts [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+      }),
+      {}
+    )
+  } catch (err) {
+    toast.error(err.message)
+    return null
+  }
+}
+
+const putSampleSizes = async (
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/useSampleSizes.ts:47]: i think this should actually be called `postRound` and go in `useRounds`

File: client/src/components/MultiJurisdictionAudit/Setup/Review/useSampleSizes.ts [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+): [
+  IStringSampleSizeOptions | null,
+  (sampleSizes: ISampleSizes) => Promise<boolean>
+] => {
+  const [
+    sampleSizeOptions,
+    setSampleSizeOptions,
+  ] = useState<IStringSampleSizeOptions | null>(null)
+
+  const uploadSampleSizes = async (sizes: ISampleSizes): Promise<boolean> => {
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/useSampleSizes.ts:89]: similar to comment above, i think this makes more sense as `createRound` as part of `useRounds`

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx (hunk 6 above)
Comment (jonahkagan) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:104]: i'm thinking something like this:
```ts
interface ISampleSizeOption {
  key: string
  size: number | null
  prob: number | null
}
```
a few examples:
```ts
{ key: "asn", size: 20, prob: 0.57 }
{ key: "prob90", size: 56, prob: 0.9 }
{ key: "custom", size: null, prob: null } // before the user sets a custom size
{ key: "custom", size: 43, prob: null } // after the user sets a custom size
```

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx [hunk 10]
```diff
@@ -1,53 +1,87 @@
...
+      body: JSON.stringify({
+        sampleSizes: { 'contest-id': 5 },
+        roundNum: 1,
+      }),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'POST',
+    },
+  },
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx:63]: one pattern i played with is making these API calls functions that take in whatever data you want to customize between tests. e.g. these three could become:

```ts
postRound: (sampleSizes) => ({
    url: '/api/election/1/round',
    response: { status: 'ok' },
    options: {
      body: JSON.stringify({
        sampleSizes,
        roundNum: 1,
      }),
      headers: {
        'Content-Type': 'application/json',
      },
      method: 'POST',
    },
  })
```

and then in the test you can say:
```ts
apiCalls.postRound({ 'contest-id': 46 })
```

dunno if it makes a big difference. curious what you think

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx [hunk 11]
```diff
@@ -81,213 +117,199 @@ beforeEach(() => {
...
-    await waitFor(() => expect(apiMock).toHaveBeenCalled())
-    expect(container).toMatchSnapshot()
+    const expectedCalls = [
+      apiCalls.getJurisdictions,
+      apiCalls.getJurisdictionFile,
+      apiCalls.getTargetedContests,
+      apiCalls.getSampleSizeOptions,
+    ]
+    await withMockFetch(expectedCalls, async () => {
+      const { container } = render(
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx:127]: could potentially use the `renderWithRouter` test helper

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx [hunk 12]
```diff
@@ -81,213 +117,199 @@ beforeEach(() => {
...
-          roundNum: 1,
-        }),
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        method: 'POST',
-      })
+      await screen.findByText('Review & Launch')
+      const launchButton = screen.getByText('Launch Audit')
+      userEvent.click(launchButton, { bubbles: true })
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx:231]: i don't think you need `{ bubbles: true }` anymore

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx [hunk 13]
```diff
@@ -46,51 +45,19 @@ const Review: React.FC<IProps> = ({ prevStage, locked, refresh }: IProps) => {
...
-          roundNum: 1,
-        }),
-        headers: {
-          'Content-Type': 'application/json',
-        },
-      })
-      // checkAndToast left here for consistency and reference but not tested since it's vestigial
-      /* istanbul ignore next */
-      if (checkAndToast(result)) {
+      if (uploadSampleSizes(sampleSizes)) {
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:55]: missing `await` here?

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx (hunk 5 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:14]: @jonahkagan Hmmm... we should look at how we're handling that in the other validations. I think it crops up more than here, and will require a solution across the board.

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx (hunk 6 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:104]: @jonahkagan That has merit, yeah. I can picture how that would work on the frontend, and it would definitely clear up that particular bug. Would the `type` field possibly also be marked as 'custom'? It might make things easier on the frontend for generating the details of the different options, but I don't know if that would make it more complex on the backend.

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx (hunk 10 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx:63]: @jonahkagan Oh yes, I remember that now. Good catch. :slightly_smiling_face:

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx (hunk 6 above)
Comment (jonahkagan) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:104]: I was thinking of getting rid of the `type` field since currently it's only getting used for `asn`. I was thinking you could use the `key` field instead. If you're into this, I'll update my PR with the modified data format

Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:104]: @jonahkagan Oh right, I missed that you moved the `asn` to the key. That makes perfect sense. Go for it! :slightly_smiling_face:

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx (hunk 7 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:209]: @jonahkagan I find the ES6 format a bit more self-documenting and readable, but that's probably just me. And this will likely change a bit when you redo the api. I'll look at it again when I refactor for that.

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx [hunk 14]
```diff
@@ -136,139 +121,163 @@ const Review: React.FC<IProps> = ({ prevStage, locked, refresh }: IProps) => {
...
+              <tr key={c.id}>
+                <td>{c.name}</td>
+                <td>{c.jurisdictionIds.join(', ')}</td>
+              </tr>
+            ))}
+          </tbody>
+        </ContestsTable>
+      </ElevatedCard>
+      <br />
+      <H4>Estimated Sample Sizes</H4>
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:193]: I feel like this header is a bit misleading. The sample sizes are the actual sizes that are going to be used, they aren't just estimated. I feel like it should say "Select Sample Size" or just "Sample Size" or something like that. @mcchilders

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx [hunk 15]
```diff
@@ -136,139 +121,163 @@ const Review: React.FC<IProps> = ({ prevStage, locked, refresh }: IProps) => {
...
       <Formik
         initialValues={{
-          sampleSize: sampleSizeOptions.length ? sampleSizeOptions[0].size : '',
-          customSampleSize: '',
+          sampleSizes: initialValues,
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:196]: does this need to be nested, or could it just be `intialValues={initialValues}` now?

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx [hunk 16]
```diff
@@ -279,7 +288,9 @@ const Review: React.FC<IProps> = ({ prevStage, locked, refresh }: IProps) => {
                   locked ||
                   !isSetupComplete(jurisdictions, contests, auditSettings)
                 }
-                onClick={handleSubmit}
+                onClick={() => {
+                  handleSubmit()
+                }}
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:293]: unnecessary change?

File: client/src/components/MultiJurisdictionAudit/Setup/Review/useSampleSizes.ts [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+import { ISampleSizeOption } from '../../../../types'
+
+export interface IStringSampleSizeOption {
+  size: string
+  key: string
+  prob: number | null
+}
+
+export interface IStringSampleSizeOptions {
+  [key: string]: IStringSampleSizeOption[]
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/useSampleSizes.ts:13]: i think it's kind of confusing to call this `key` because `key` refers to something different in `IStringSampleSizeOption`. I think `contestId` would be clearer. same feedback for the other types below

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx [hunk 18]
```diff
@@ -1,53 +1,60 @@
...
+    },
+  }),
+  getEmptyJurisdictions: {
+    url: '/api/election/1/jurisdiction',
+    response: { jurisdictions: [] },
+  },
+  getJurisdictions: {
+    url: '/api/election/1/jurisdiction',
+    response: { jurisdictions: jurisdictionMocks.allManifests },
+  },
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx:44]: could combine these like you did with `postRound`. same with the `getContests` calls below

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx [hunk 19]
```diff
@@ -81,213 +98,198 @@ beforeEach(() => {
...
-        body: JSON.stringify({
-          sampleSize: 46,
-          roundNum: 1,
-        }),
-        headers: {
-          'Content-Type': 'application/json',
-        },
-        method: 'POST',
-      })
+    auditSettingsMock.mockReturnValue(settingsMock.full)
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx:190]: any reason this is mocked while all the other data sources use the mock API?

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx [hunk 20]
```diff
@@ -81,213 +98,198 @@ beforeEach(() => {
...
+    const expectedCalls = [
+      apiCalls.getJurisdictions,
+      apiCalls.getJurisdictionFile,
+      apiCalls.getTargetedContests,
+      apiCalls.getSampleSizeOptions,
+      apiCalls.postRound({ 'contest-id': 46 }),
+      apiCalls.getSampleSizeOptions,
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx:197]: surprised that there's another call to `getSampleSizeOptions` after launching the round

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx [hunk 21]
```diff
@@ -81,213 +98,198 @@ beforeEach(() => {
...
+    ]
+    await withMockFetch(expectedCalls, async () => {
+      renderView()
+      await screen.findByText('Review & Launch')
+      const launchButton = screen.getByText('Launch Audit')
+      userEvent.click(launchButton)
+      await screen.findByText('Are you sure you want to launch the audit?')
+      const confirmLaunchButton = screen.getAllByText('Launch Audit')[1]
+      userEvent.click(confirmLaunchButton)
+      await waitFor(() => expect(refreshMock).toHaveBeenCalled())
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx:207]: another option would be to find something on the next screen that indicates you were successfully redirected/the audit got launched. i suggest that because it's more in line with the philosophy of testing from the user's perspective

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx [hunk 22]
```diff
@@ -81,213 +98,198 @@ beforeEach(() => {
...
+      apiCalls.getSampleSizeOptions,
+    ]
+    await withMockFetch(expectedCalls, async () => {
+      renderView()
+      const newSampleSize = await screen.findByText(
+        'Enter your own sample size (not recommended)'
+      )
+      userEvent.click(newSampleSize)
+      const customSampleSizeInput = await screen.findByRole('textbox')
+      fireEvent.change(customSampleSizeInput, { target: { value: '40' } }) // userEvent has a problem with this field due to the lack of an explicit value field: https://github.com/testing-library/user-event/issues/356
```
Comment (jonahkagan) [inline client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx:273]: that's annoying. why doesn't it have a value field? isn't it a regular `input`?

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx (hunk 15 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:196]: @jonahkagan Formik needs a root object to name things, and if we made that the undeterministic contest ids, that would make things more confusing for referencing within the code, so I think this is the best way to do it.

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx (hunk 16 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.tsx:293]: @jonahkagan Oh good catch. I was adding in a console.log there to check that it was submitting when I was testing things, but forgot to fix it all the way when I removed the log.

File: client/src/components/MultiJurisdictionAudit/Setup/Review/useSampleSizes.ts (hunk 17 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/useSampleSizes.ts:13]: @jonahkagan I actually forgot you could use something other than `key` there, haha.

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx (hunk 22 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx:273]: @jonahkagan It's wrapped a bunch of times, but ultimately it's the [input group element from Blueprint](https://blueprintjs.com/docs/#core/components/text-inputs). And from examining how they handle things, it looks like the fields don't have a value attribute until you start entering data into them. Which would explain why userEvent is getting confused.

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx (hunk 21 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx:207]: @jonahkagan It was either check that the mock for `refresh` was called, or that the mock for `history.push` was called. The former just seemed simpler. Both happen when it is successfully launched, though.

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx (hunk 20 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx:197]: @jonahkagan That's due to me copy-pasting the pattern from one of the other api hooks, where you reload the data from the server after posting the update. In this context, though, I don't think that's necessary, so should change that probably.

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx (hunk 19 above)
Comment (MorganLove) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx:190]: @jonahkagan I think that's a holdover from the old tests, before mocking the api was simplified. The settings were tested directly, and adding those mocks into the mix would have made a horrible mess. We could integrate the settings into the sample size api mocks now, I think, though this is still cleaner imo.

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx (hunk 20 above)
Comment (jonahkagan) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx]: oh interesting, `withMockFetch` is supposed to fail if its given expected API calls that don't occur

File: client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx (hunk 21 above)
Comment (jonahkagan) [reply client/src/components/MultiJurisdictionAudit/Setup/Review/index.test.tsx:210]: oh right, forgot there were no other routes to navigate to in this test setup
